### DISCUSSION TITLE
Dynamically hiding column makes table flicker

### DISCUSSION THREAD

**Discussion by yngwi** [TanStack/table#3859] [Q&A] [answered]
Hi, I'm having a table in v8 (alpha 48) where a user can toggle between normal and editing mode with a button. If using the Editing mode, an additional column (_actions_) is visible where rows can be deleted and other actions performed". This is hidden in the normal view. The hiding is done with the following part of the table instance configuration:

```js
state: {
  columnVisibility: editing ? {} : { actions: false },
},
```
Now, whenever the "Editing" button is clicked, the table "flickers" once and the scrollbar is visible very shortly while the additional column is added / removed. Is there a way to avoid this? I'm not sure if I do this correctly, currently, the "useTableInstance" is receiving this "editing" state in its configuration.  I have also tried using a useEffect hook that calls instance.toggleColumnVisibility() for this column depending on the "editing" variable, but the result is the same.

I'm grateful for any pointers.
Best,
Daniel

**Comment by tannerlinsley** [comment]
CodeSandBox showing this issue would really really help diagnose it

**Reply by yngwi** [reply]
Hi, I have tried to reproduce on [StackBlitz](https://stackblitz.com/edit/vitejs-vite-zvf5qc?file=src%2FApp.tsx) (I hope that's OK) but here it seems to work as expected. In my original repository I have been using the "Mantine" UI library to style the table and react router 6 with nested routes, maybe something there is actually causing the problem. I'm going to investigate...

**Reply by yngwi** [reply]
Ok it seems indeed to be a problem with a specific UI component. Sorry for the unneeded question and thanks for your feedback.

**Reply by yngwi** [reply]
More specifically (FYI), it was a problem with Mantine [Popover](https://mantine.dev/core/popover/?t=props) components. They render content in a [Portal](https://mantine.dev/core/portal/) which triggered the page resizing. Setting it to not use portals with the `withinPortal={false}` prop fixes this.

**Comment by yngwi** [comment, marked as answer]
Fixed by modifying the UI library component usage, see here: https://github.com/TanStack/react-table/discussions/3859#discussioncomment-2641747
